src/util/waitForElement.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Waits until Element will appear
+ *
+ * @api private
+ * @method _waitForElement
+ * @param {string} elSelector Selector to locate Element
+ * @param {() => void} callback Callback to be called after Element appearance
+ */
+export default function _waitForElement(elSelector, callback) {
+  if (document.querySelector(elSelector) !== null) {
+    callback();
+    return;
+  }
+
+  if (typeof MutationObserver !== "undefined") {
+    /* jshint ignore:start */
+    const observer = new MutationObserver(() => {
+      if (document.querySelector(elSelector) !== null) {
+        observer.disconnect();
+        callback();
+      }
+    });
+
+    observer.observe(document.body, {
+      childList: true,
+      subtree: true,
+      attributes: false,
+      characterData: false,
+    });
+    /* jshint ignore:end */
+  } else {
+    // Old browsers will wait by timeout
+    waitForElementByTimeout(elSelector, callback, 1000, 10000);
+  }
+}
+
+/**
+ * @param {string} elSelector
+ * @param {() => void} callback
+ * @param {number} checkInterval In milliseconds
+ * @param {number} maxTimeout In milliseconds
+ */
+function waitForElementByTimeout(
```

Comment:
Can we test this function please? I'm a little concerned since it's doing a recursive call.

Reply:
Added tests in https://github.com/gooddaytoday/intro.js/commit/d5eaf0d3d2f29f3fc6d934c1e99190954b48181a